### DISCUSSION TITLE
Update doesn't work via model method

### DISCUSSION THREAD

**Discussion by andrewebdev** [charmbracelet/bubbletea#474] [Q&A] [answered]
I'm having an issue where I'm obviously doing something wrong, or misunderstanding something.
If I change a value on a model directly, the view is updated correctly. However, if I try the same thing, via a method on the model interface, then nothing happens. The `focussed` attribute never changes.

See below:

_Snippet from update()_
```golang
		case key.Matches(msg, keys.Next):
			// m.focussed += 1   // This works
			cmd = m.Next()       // This does NOT work
```

_Snippet from my model interface_
```golang
func (m Model) Next() tea.Cmd {
	if m.focussed < 2 {
		m.focussed += 1
	} else {
		m.focussed = 0
	}
	return nil
}
```

**Comment by muesli** [comment, marked as answer]
```go
func (m Model) Next() tea.Cmd {
...
}
```

This function is working on a copy of your `Model`. If you change it to a pointer receiver, it will correctly update:

```go
func (m *Model) Next() tea.Cmd {
...
}
```

**Reply by andrewebdev** [reply]
Eish, my hand isn't big enough for the facepalm I deserve. That was a rookie mistake.

**Reply by muesli** [reply]
Don't worry, we've all been there :smiley:
